apps/dg/views/component_view.js

```diff
@@ -483,15 +486,31 @@ DG.ComponentView = SC.View.extend(
                   tX = DG.ViewUtilities.roundToGrid(info.left + (evt.pageX - info.pageX)),
                   tY = DG.ViewUtilities.roundToGrid(info.top + (evt.pageY - info.pageY)),
                   tContainerWidth = this.getContainerWidth(),
-                  tContainerHeight = this.getContainerHeight();
+                  tContainerHeight = this.getContainerHeight(),
+                  tMinX = -info.width + kMinSize,
+                  tMaxX = tContainerWidth - kMinSize,
+                  tMinY = -kTitleBarHeight / 2,
+                  tMaxY = tContainerHeight - kTitleBarHeight / 2;
+              if (DG.KEEP_IN_BOUNDS_PREF) {
+                var tInspectorDimensions = tOuterView.getInspectorDimensions();
+                tMinX = 0;
+                tMaxX = tContainerWidth - (info.width + tInspectorDimensions.width);
+                tMinY = 0;
+                tMaxY = tContainerHeight - (Math.max(info.height, tInspectorDimensions.height));
+              }
 
-              tX = Math.min(Math.max(tX, -info.width + kMinSize),
-                  tContainerWidth - kMinSize);
+              tX = Math.min(Math.max(tX, tMinX), tMaxX);
               tOuterView.adjust('left', tX);
 
-              tY = Math.min(Math.max(tY, -kTitleBarHeight / 2),
-                  tContainerHeight - kTitleBarHeight / 2);
+              tY = Math.min(Math.max(tY, tMinY), tMaxY);
               tOuterView.adjust('top', tY);
+
+              if (DG.KEEP_IN_BOUNDS_PREF) {
```

Comment:
There's a lot of repetitive code in these `KEEP_IN_BOUNDS_PREF` clauses from here on. Any thoughts about refactoring to reduce bulk?

Reply:
I will take a look at this.  Probably can be improved.